### PR TITLE
Fix #3653: Bazel Build Fail for MacOS

### LINKED ISSUE #3653: Bazel Build Fail for MacOS
Original issue - https://github.com/protocolbuffers/protobuf/issues/8863

### PR DESCRIPTION
**Pull request by anandwana001**
## Explanation

Fix #3653 

Original Issue - https://github.com/protocolbuffers/protobuf/issues/8863

Screenshot 2021-08-07 at 09 59 23

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: third_party/versions.bzl [hunk 1]
```diff
@@ -106,12 +106,16 @@ MAVEN_TEST_DEPENDENCY_VERSIONS = {
 
 # Note to developers: Please keep this dict sorted by key to make it easier to find dependencies.
 HTTP_DEPENDENCY_VERSIONS = {
+    "bazel_skylib": {
+        "sha": "97e70364e9249702246c0e9444bccdc4b847bed1eb03c5a3ece4f83dfe6abc44",
+        "version": "1.0.2",
```
Comment (BenHenning) [inline third_party/versions.bzl:111]: Why not use the latest release?

File: third_party/versions.bzl [hunk 2]
```diff
@@ -106,12 +106,16 @@ MAVEN_TEST_DEPENDENCY_VERSIONS = {
...
+    },
     "dagger": {
         "sha": "9e69ab2f9a47e0f74e71fe49098bea908c528aa02fa0c5995334447b310d0cdd",
         "version": "2.28.1",
     },
     "protobuf_tools": {
-        "version": "3.11.0",
+        "version": "3.12.0",
```
Comment (BenHenning) [inline third_party/versions.bzl:118]: Could this be the latest protoc?

**Review by BenHenning — COMMENTED** [review]
Thanks @anandwana001. Nice find. I wish that we would understand _why_ this change works, but it might be something as simple as the old dependencies aren't built against the latest runtime libraries on OS X.

Is it possible for us to add an OS X flavor to https://github.com/oppia/oppia-android/blob/develop/.github/workflows/build_tests.yml#L19 so that we can verify that this works? I think we need to verify that it first fails without the fixes, but then that the fixes fix the OS X build (which is specifically convenient since it provides a regression test).

**Comment by anandwana001** [comment]
Putting this PR on less priority for now, will check again in next week.

**Pull request closed without merging**